### PR TITLE
INT B-18916 create customer and okta profile as office user (MERGE AFTER B-18915)

### PR DESCRIPTION
**Pull request by danieljordan-caci**
## Summary

This PR does contain the code of a currently open PR to use the customer form to ensure functionality. 
[B-18915](https://github.com/transcom/mymove/pull/12248)

A requirement of [E-04903](https://www13.v1host.com/USTRANSCOM38/Epic.mvc/Summary?oidToken=Epic:888544) is for a SC to be able to create a MilMove account for a customer. 

Using the form in the PR above, we can call the backend to do three things:
1. Create the user
2. Create the service member
3. Create the Okta profile (if chosen to do so)

This PR does the following:
- creates a new GHC API endpoint, `POST /createCustomerWithOktaOption`
- adds okta user creation API url to `provider.go`
- adds handler for `createCustomerWithOktaOption` endpoint
- adds function within handler to create the okta user
- adds model to payload conversion in `model_to_payload.go`
- add Okta Go models for profile creation
- minor UI formatting and changes to better handle backend changes
- add flash message to search page and display on successful creation of a customer

### How to test

1. Fire up `client` and `server`
2. Login as a service counselor
3. Search for a user that doesn't exist and click the `Add Customer` button at the bottom of the page
4. You should now see the form to create a customer (click "yes" for okta account creation)
5. Fill it out and call the API (prob best to have your debugger turned on)
6. Make sure all values are being passed into the backend and go through the flow
7. An okta account should now be created and you should get an activation email if you were using the `user+variation@gmail.com` method
8. Log out as a service counselor
9. Log into MilMove as a customer and click the `Local Sign-in` option
10. You should now see the email for the user you just created in the list
11. You can sign in as that user and create a move if ya want

### CODE REVIEW

**Review by traskowskycaci — APPROVED** [review]
Saw everything come through as expected on the backend, got the email, was able to login as that service member and create moves. Very nice!

**Review by cameroncaci — CHANGES_REQUESTED** [review]
Tested perfectly like a charm. See suggested code changes below

File: pkg/handlers/ghcapi/customer.go [hunk 1]
```diff
@@ -86,3 +110,202 @@ func (h UpdateCustomerHandler) Handle(params customercodeop.UpdateCustomerParams
...
+			// now we will take all the data we have and build the service member
+			userID := user.ID
+			residentialAddress := addressModelFromPayload(&payload.ResidentialAddress.Address)
+			backupMailingAddress := addressModelFromPayload(&payload.BackupMailingAddress.Address)
+
+			// Create a new serviceMember using the userID
+			newServiceMember := models.ServiceMember{
+				UserID:               userID,
+				Edipi:                payload.Edipi,
+				Affiliation:          (*models.ServiceMemberAffiliation)(payload.Affiliation),
```
[the remark this answers is not in the document]
Comment (danieljordan-caci) [reply pkg/handlers/ghcapi/customer.go:163]: `edipi` is a pointer in this `ServiceMember` struct:
```
type ServiceMember struct {
	ID                     uuid.UUID                 `json:"id" db:"id"`
	CreatedAt              time.Time                 `json:"created_at" db:"created_at"`
	UpdatedAt              time.Time                 `json:"updated_at" db:"updated_at"`
	UserID                 uuid.UUID                 `json:"user_id" db:"user_id"`
	User                   User                      `belongs_to:"user" fk_id:"user_id"`
	Edipi                  *string                   `json:"edipi" db:"edipi"`
	Affiliation            *ServiceMemberAffiliation `json:"affiliation" db:"affiliation"`
	FirstName              *string                   `json:"first_name" db:"first_name"`
	MiddleName             *string                   `json:"middle_name" db:"middle_name"`
	LastName               *string                   `json:"last_name" db:"last_name"`
	Suffix                 *string                   `json:"suffix" db:"suffix"`
	Telephone              *string                   `json:"telephone" db:"telephone"`
	SecondaryTelephone     *string                   `json:"secondary_telephone" db:"secondary_telephone"`
	PersonalEmail          *string                   `json:"personal_email" db:"personal_email"`
	PhoneIsPreferred       *bool                     `json:"phone_is_preferred" db:"phone_is_preferred"`
	EmailIsPreferred       *bool                     `json:"email_is_preferred" db:"email_is_preferred"`
	ResidentialAddressID   *uuid.UUID                `json:"residential_address_id" db:"residential_address_id"`
	ResidentialAddress     *Address                  `belongs_to:"address" fk_id:"residential_address_id"`
	BackupMailingAddressID *uuid.UUID                `json:"backup_mailing_address_id" db:"backup_mailing_address_id"`
	BackupMailingAddress   *Address                  `belongs_to:"address" fk_id:"backup_mailing_address_id"`
	Orders                 Orders                    `has_many:"orders" fk_id:"service_member_id" order_by:"created_at desc" `
	BackupContacts         BackupContacts            `has_many:"backup_contacts" fk_id:"service_member_id"`
}
```

[the remark this answers is not in the document]
Comment (cameroncaci) [reply pkg/handlers/ghcapi/customer.go:163]: Yup, you're right. Apologies, disregard this one. I had created users with empty values and automatically assumed this was a static string not pointer

File: swagger-def/ghc.yaml [hunk 2]
```diff
@@ -3394,7 +3430,66 @@ definitions:
...
+      residentialAddress:
+        $ref: 'definitions/Address.yaml'
+      backupContact:
+        $ref: '#/definitions/BackupContact'
+      id:
+        type: string
+        format: uuid
+        example: c56a4180-65aa-42ec-a945-5fd21dec0538
+      edipi:
+        type: string
```
[the remark this answers is not in the document]
Comment (cameroncaci) [reply swagger-def/ghc.yaml:3473]: Somewhere allows the creation of "empty" EDIPIs, which would be good to set back to null

File: pkg/handlers/ghcapi/customer.go [hunk 3]
```diff
@@ -86,3 +110,202 @@ func (h UpdateCustomerHandler) Handle(params customercodeop.UpdateCustomerParams
...
+			defaultPermission := models.BackupContactPermissionEDIT
+			backupContact, verrs, err := newServiceMember.CreateBackupContact(appCtx.DB(),
+				*payload.BackupContact.Name,
+				*payload.BackupContact.Email,
+				payload.BackupContact.Phone,
+				models.BackupContactPermission(defaultPermission))
+			if err != nil || verrs.HasAny() {
+				return handlers.ResponseForVErrors(appCtx.Logger(), verrs, err), err
+			}
+
```
[the remark this answers is not in the document]
Comment (danieljordan-caci) [reply pkg/handlers/ghcapi/customer.go]: Added

File: pkg/handlers/ghcapi/customer.go [hunk 4]
```diff
@@ -86,3 +110,202 @@ func (h UpdateCustomerHandler) Handle(params customercodeop.UpdateCustomerParams
...
+	if err != nil {
+		appCtx.Logger().Error("error marshaling payload", zap.Error(err))
+		return nil, err
+	}
+
+	// making HTTP request to Okta Users API to create a user
+	// this is done via a POST request for creating a user that sends an activation email (when activate=true)
+	// https://developer.okta.com/docs/reference/api/users/#create-user-without-credentials
+	req, _ := http.NewRequest("POST", baseURL, bytes.NewReader(body))
+	h := req.Header
```
[the remark this answers is not in the document]
Comment (danieljordan-caci) [reply pkg/handlers/ghcapi/customer.go]: Added

File: swagger-def/ghc.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (danieljordan-caci) [reply swagger-def/ghc.yaml:3472]: Made this nullable

File: swagger-def/ghc.yaml [hunk 5]
```diff
@@ -3438,7 +3533,59 @@ definitions:
...
+      secondaryTelephone:
+        type: string
+        format: telephone
+        pattern: '^[2-9]\d{2}-\d{3}-\d{4}$'
+        x-nullable: true
+      personalEmail:
+        type: string
+        format: x-email
+        example: personalEmail@email.com
+        x-nullable: true
```
[the remark this answers is not in the document]
Comment (danieljordan-caci) [reply swagger-def/ghc.yaml]: Added

**Review by cameroncaci — APPROVED** [review]
beautiful

**Pull request merged**